### PR TITLE
The following changes have been done:

### PR DESCRIPTION
**Pull request by mdpradeep**
1) Removed HTTPClient dependency by moving to Requests.jl
    2) Removed dependency on LibExpat by moving to LightXML.jl
    3) Implementation of SQS APIs
    4) Documentation update

### CODE REVIEW

**Comment by mdpradeep** [comment]
@ViralBShah

File: REQUIRE [hunk 1]
```diff
@@ -1,6 +1,7 @@
 julia 0.4-
```
Comment (tkelman) [inline REQUIRE]: remove the trailing `-`

**Comment by tkelman** [comment]
This could likely use some testing from existing users of the package, since it's a pretty major rewrite. Would certainly want to change the minor version number for the first tag after merging this.

**Comment by ViralBShah** [comment]
Not sure how to locate users to ask them to test. I believe that reasonably, testing will only happen after tagging.

**Comment by ViralBShah** [comment]
Would be great if @amitmurthy and @tanmaykm can have a look.

**Comment by amitmurthy** [comment]
Nice work. LGTM.

All tests passing locally?

**Comment by mdpradeep** [comment]
Yes, all tests in test/test*.jl are passing.

Regards,
Pradeep

**Comment by tkelman** [comment]
Add a `test/runtests.jl` file so `Pkg.test` can work?

**Comment by amitmurthy** [comment]
@tkelman that would require AWS credentials. All the tests actually use AWS resources and are charged.

**Comment by mdpradeep** [comment]
I can, but it requires AWS credentials. I could create a separate id, but then anyone would be able to launch VMs, etc… So, not sure whether this is a good idea or not.

Regards,
Pradeep

**Comment by amitmurthy** [comment]
No, not a good idea to make the credentials public.

**Comment by tkelman** [comment]
Travis has features where you can encrypt secure env vars. Depends on the resource cost per time the tests run whether it's worth setting up. Would Pkg.test fail immediately if you don't have credentials set up?

**Comment by amitmurthy** [comment]
Yes. The EC2 tests check if servers can be booted up using the AWS API, the S3 tests if buckets can be created and accessed and so on.

Travis encryption features may be worth looking into.

**Comment by ViralBShah** [comment]
I think the resource costs will be tiny, and we can probably do it with the free tier on a new ID.

**Comment by amitmurthy** [comment]
We can merge this now while we explore options for safely passing credentials to Travis.

File: REQUIRE (hunk 1 above)
Comment (amitmurthy) [inline REQUIRE]: @mdpradeep please update this and we can merge.

**Comment by mdpradeep** [comment]
Updated

**Comment by amitmurthy** [comment]
Can you squash the commits too?

**Comment by mdpradeep** [comment]
Done. Let me know if this is fine.

**Comment by ViralBShah** [comment]
Should we tag a new release, and bump the minor number?

**Comment by tkelman** [comment]
Hard to say whether the tests here are representative of real world use cases, but when this gets tagged it should absolutely change at least the minor version number.

**Pull request merged**